lib/type/boolean.ex

```diff
@@ -1,15 +1,52 @@
 defmodule Litmus.Type.Boolean do
-  @moduledoc """
-  Schema and validation for Boolean data type.
-  """
+  @moduledoc false
 
-  defstruct truthy: [true, "true"],
-            falsy: [false, "false"],
+  alias Litmus.Required
+
+  @truthy_default [true, "true"]
+  @falsy_default [false, "false"]
+
+  defstruct truthy: @truthy_default,
+            falsy: @falsy_default,
             required: false
 
   @type t :: %__MODULE__{
           truthy: [term],
           falsy: [term],
           required: boolean
         }
+
+  @spec validate_field(t, binary, map) :: {:ok, map} | {:error, binary}
+  def validate_field(type, field, data) do
+    with {:ok, data} <- Required.validate(type, field, data),
+         {:ok, data} <- truthy_falsy_validate(type, field, data) do
+      {:ok, data}
+    else
+      {:error, msg} -> {:error, msg}
+    end
+  end
+
+  @spec truthy_falsy_validate(t, binary, map) :: {:ok, map} | {:error, binary}
+  defp truthy_falsy_validate(%__MODULE__{falsy: falsy, truthy: truthy}, field, params) do
+    cond do
+      !Map.has_key?(params, field) ->
+        {:ok, params}
+
+      params[field] in Enum.uniq(truthy ++ @truthy_default) ->
```

Comment:
do we need to check case sensitivity on the truthy value? e.g. if truthy is ["true"], do we need to account for inputs like "tRue"?

Reply:
I hadn't considered that case. Joi supports it. I think we should too. What does everyone think? @lob/atlas